test/format/import.c

```diff
@@ -0,0 +1,200 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "format.h"
+
+static int cursor_insert(WT_CURSOR *cursor, int key, int val);
+static void copy_file(WT_SESSION *session, const char *name);
+
+/*
+ * Import directory initialize command, remove and re-create the primary backup directory, plus a
+ * copy we maintain for recovery testing.
+ */
+#define HOME_IMPORT_INIT_CMD "rm -rf %s/IMPORT && mkdir %s/IMPORT"
+#define IMPORT_URI "table:import"
+#define IMPORT_URI_CONFIG "key_format=i,value_format=i"
+/*
+ * import --
+ *     Periodically import table.
+ */
+WT_THREAD_RET
+import(void *arg)
+{
+    WT_CONNECTION *conn, *import_conn;
+    WT_CURSOR *cursor, *metadata_cursor;
+    WT_DECL_RET;
+    WT_SESSION *import_session, *session;
+    u_int period;
+    u_int64_t counter;
+    char buf[2048], *cmd;
+    const char *table_config, *file_config;
+    size_t len;
+    bool import_value;
+
+    (void)(arg);
+    conn = g.wts_conn;
+    import_value = false;
+    counter = 0;
+
+    len = strlen(g.home) * 2 + strlen(HOME_IMPORT_INIT_CMD) + 1;
+    cmd = dmalloc(len);
+    testutil_check(__wt_snprintf(cmd, len, HOME_IMPORT_INIT_CMD, g.home, g.home));
+    testutil_checkfmt(system(cmd), "%s", "import directory creation failed");
+    free(cmd);
+
+    len = strlen(g.home) + strlen("IMPORT") + 10;
+    cmd = dmalloc(len);
+    printf("len %lu\n", len);
+    testutil_check(__wt_snprintf(cmd, len, "%s/IMPORT", g.home));
+    /* Open a connection to the database, creating it if necessary. */
+    testutil_check(wiredtiger_open(cmd, NULL, "create", &import_conn));
+    free(cmd);
+
+    /* Open a session */
+    testutil_check(import_conn->open_session(import_conn, NULL, NULL, &import_session));
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+    testutil_checkfmt(
+      import_session->create(import_session, IMPORT_URI, IMPORT_URI_CONFIG), "%s", g.uri);
+
+    while (!g.workers_finished) {
+        period = mmrand(NULL, 1, 10);
+        /*
+         * open_cursor can return EBUSY if concurrent with a metadata operation, retry in that case.
+         */
+        while ((ret = import_session->open_cursor(
+                  import_session, IMPORT_URI, NULL, NULL, &cursor)) == EBUSY)
+            __wt_yield();
+        testutil_check(ret);
+
+        // Check if insert operation is re-usable from test/format
+        // Validate the cursor inserts
+        for (int i = 0; i < 1000; ++i, ++counter) {
+            testutil_check(cursor_insert(cursor, counter + (uint64_t)i, i));
```

Comment:
Use update here, to keep file size low. make key = i, and value = counter + i instead.